app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedFragment.kt

```diff
@@ -25,18 +30,27 @@ class FeedFragment : Fragment() {
     private lateinit var dataBinding: FeedFragmentBinding
     private lateinit var viewModel: FeedViewModel
     private lateinit var feedAdapter: FeedAdapter
+    private lateinit var feedRepository: FeedRepository
 
     override fun onCreateView(inflater: LayoutInflater,
                               container: ViewGroup?,
                               savedInstanceState: Bundle?): View? {
         dataBinding = DataBindingUtil.inflate(inflater, R.layout.feed_fragment, container, false)
+        setupRepository()
         setupViewModel()
         setupViews()
         return dataBinding.root
     }
+    private fun setupRepository()
+    {
+        val feedDatabase = AppDatabase.createDatabase(requireActivity().application)
+        val service = ServiceBuilder.buildService(FeedApiService::class.java)
+
+        feedRepository = FeedRepository(service, feedDatabase)
```

Comment:
And we don't want to recreate the repository every time the fragment is created. So, let's move this line to `FeedViewModelFactory`. It will be called only when no `FeedViewModel` exists.